code-styling/mysql/README.md

```diff
@@ -0,0 +1,574 @@
+
+SELECT
+  t1.id,
+  t1.column_a
+FROM table1 AS t1
+LEFT JOIN table2 AS t2
+ON t1.id = t2.table1_id
+AND t2.table1_id IS NULL;
+
+SELECT
+  t1.id,
+  t1.column_a
+FROM table1 AS t1
+LEFT JOIN table2 AS t2
+  ON t1.id = t2.table1_id AND t2.table1_id IS NULL;
+```
+
+### Delete queries
+
+Delete queries SHOULD not be used.
+
+If a delete query is required, the table from which the rows are deleted MUST be
+specified on the same line as the `DELETE FROM` keyword. Delete queries also
+MUST specify a WHERE clause.
+
+Good delete queries:
+
+```sql
+DELETE FROM my_table
+WHERE id = 5;
+```
+
+Bad delete queries:
+
+```sql
+-- Invalid: `FROM` must be on same line as `DELETE`
+DELETE
+FROM my_table
+WHERE id = 5;
+
+-- Invalid: table name must be on same line as `FROM`
+DELETE FROM
+my_table
+WHERE id = 5;
+
+-- Where clause should follow `WHERE` keyword
+DELETE FROM my_table
+WHERE
+  id = 5;
+
+-- Missing `WHERE` clause
+DELETE FROM my_table;
+```
+
+### Insert queries
+
+Insert queries MUST specify all columns in which data will be inserted. Omitted
+columns MUST have a default value. The `id` column MUST NOT be specified in
+insert queries unless those queries are used in some kind of data replication.
+
+If an insert query specifies more than 5 columns, the `SET` syntax is
+RECOMMENDED. In that case the `SET` keyword MUST follow the table name and the
+column names MUST be indented one level.
+
+Insert queries MAY specify more than one row.
+
+Good insert queries:
+
+```sql
+INSERT INTO my_table (column_a, column_b)
+VALUES ('A', 'B');
+
+INSERT INTO my_table SET
+  column_a = 'A',
+  column_with_long_name = 'B',
+  column_with_even_longer_name = 'C',
+  column_d = 'D';
+
+INSERT INTO my_table (
+  column_a,
+  column_b
+) VALUES (
+  'A',
+  'B'
+);
+
+INSERT INTO my_table (column_a, column_b)
+VALUES
+  ('A', 'B'),
+  ('C', 'D');
+```
+
+Bad insert queries:
+
+```sql
+INSERT INTO my_table (column_a, column_b) VALUES ('A', 'B');
+
+INSERT INTO my_table (column_a, column_b) VALUES ('A', 'B'), ('C', 'D');
+```
+
+### Update queries
+
+Update queries MUST have a `WHERE` clause that limits the number of rows that
+will be updated. The where clause MUST NOT be indented.
+
+The `UPDATE` keyword MUST be followed by the table name that will be affected.
+The `SET` keyword MUST be on the next line and and MUST not be indented. Column
+names should be on separate lines, followed by an equals sign and the new value
+(all on the same line). If only one column will be updated, this MAY be on the
+same line as the `SET` keyword.
+
+Good update queries:
+
+```sql
+UPDATE my_table
+SET column_a = 'new_value'
+WHERE id = 5;
+
+UPDATE my_table
+SET
+  column_a = 'new_value'
+WHERE id = 5;
+
+UPDATE my_table
+SET
+  column_a = 'new_value',
+  column_b = 'other_value'
+WHERE id = 5;
+```
+
+Bad update queries:
+
+```sql
+UPDATE my_table
+SET column_a = 'new_value', column_b = 'other_value'
+WHERE id = 5;
+```
+
+### Other select criteria
```

Comment:
It seems to me that all of these (Between, Grouping, Ordering, and Limit) should be underneath the "Select queries" section.